### PR TITLE
Weekly PR

### PR DESCRIPTION
**Pull request by EugeneButusov**
+ Feature #152 Add Leaderboards functionality to GameCetner

### CODE REVIEW

**Comment by EugeneButusov** [comment]
@borisspektor @refaelos, it's ready to merge.

File: src/documents/cocos2dx/cpp/profile/Profile_Events.html.md [hunk 1]
```diff
@@ -557,4 +557,211 @@ void Example::EVENT_INVITE_CANCELLED(EventCustom *event) {
 
   // ... your game specific implementation here ...
 }
+```
+
+### GET_LEADERBOARDS_STARTED
+
+The event `EVENT_GET_LEADERBOARDS_STARTED` is triggered when an invitation process has started.
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_Events.html.md]: @EugeneButusov it's not invitation :)

File: src/documents/cocos2dx/cpp/profile/Profile_Events.html.md [hunk 2]
```diff
@@ -557,4 +557,211 @@ void Example::EVENT_INVITE_CANCELLED(EventCustom *event) {
...
+  CCProvider provider = CCProvider((dynamic_cast<__Integer *>(eventData->objectForKey(CCProfileConsts::DICT_ELEMENT_PROVIDER)))->getValue());    
+  __String *payload = dynamic_cast<__String *>(eventData->objectForKey(CCProfileConsts::DICT_ELEMENT_PAYLOAD));
+
+  // ... your game specific implementation here ...
+}
+```
+
+### GET_LEADERBOARDS_FINISHED
+
+The event `EVENT_GET_LEADERBOARDS_FINISHED` is triggered when an leaderboards fetching has finished successfully.
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_Events.html.md]: @EugeneButusov please change to "when fetching leaderboards has..." everywhere

File: src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md [hunk 3]
```diff
@@ -41,28 +41,73 @@ This class represents a profile of a user from a social network (provider).
...
+	
     - **expiration_date** - *UNIX timestamp as cocos2dx::__Double* - `not available for Android`
 
+## CCLeaderboard <a href="https://github.com/soomla/cocos2dx-profile/blob/master/Soomla/domain/CCLeaderboard.h" target="_blank"><img class="link-icon" src="/img/tutorial_img/linkImg.png"></a>
+
+This class holds information about a leaderboard which was returned by `getLeaderboards`.
+
+**A `CCLeaderboard` contains the following elements:**
+
+- `Provider`: provider leaderboard related to
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md]: @EugeneButusov please change to "the provider this leaderboard is related to"

File: src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md [hunk 4]
```diff
@@ -41,28 +41,73 @@ This class represents a profile of a user from a social network (provider).
...
+
+## CCScore <a href="https://github.com/soomla/ios-profile/blob/master/SoomlaiOSProfile/game-services/domain/Score.h" target="_blank"><img class="link-icon" src="/img/tutorial_img/linkImg.png"></a>
+
+This class holds information about a score which was returned by `getScores` or `reportScore` methods.
+
+**A `CCScore` contains the following elements:**
+
+- `Leaderboard`: leaderboard containing this score
+
+- `Rank`: a position of this score in the leaderboard
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md]: @EugeneButusov please change "a position..." to "the position..."

File: src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md [hunk 5]
```diff
@@ -41,28 +41,73 @@ This class represents a profile of a user from a social network (provider).
...
+
+- `Player`: an owner of this score
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md]: @EugeneButusov please change "an owner..." to "the owner..."

File: src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md [hunk 6]
```diff
@@ -41,28 +41,73 @@ This class represents a profile of a user from a social network (provider).
...
+
+- `Value`: a value of this score
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md]: @EugeneButusov also here - a -> the

File: src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md [hunk 7]
```diff
@@ -41,28 +41,73 @@ This class represents a profile of a user from a social network (provider).
...
 
+## CCLeaderboard <a href="https://github.com/soomla/cocos2dx-profile/blob/master/Soomla/domain/CCLeaderboard.h" target="_blank"><img class="link-icon" src="/img/tutorial_img/linkImg.png"></a>
+
+This class holds information about a leaderboard which was returned by `getLeaderboards`.
+
+**A `CCLeaderboard` contains the following elements:**
+
+- `Provider`: provider leaderboard related to
+
+- `ID`: an unique identifier of leaderboard
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md]: @EugeneButusov "an" -> "a"

File: src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md [hunk 8]
```diff
@@ -41,28 +41,73 @@ This class represents a profile of a user from a social network (provider).
...
+
+## CCScore <a href="https://github.com/soomla/ios-profile/blob/master/SoomlaiOSProfile/game-services/domain/Score.h" target="_blank"><img class="link-icon" src="/img/tutorial_img/linkImg.png"></a>
+
+This class holds information about a score which was returned by `getScores` or `reportScore` methods.
+
+**A `CCScore` contains the following elements:**
+
+- `Leaderboard`: leaderboard containing this score
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md]: @EugeneButusov add "the" before leaderboard

File: src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md [hunk 9]
```diff
@@ -566,6 +611,100 @@ soomla::CCSoomlaProfile::getInstance()->multiShare(
 );
 ```
 
+### `getLeaderboards`
+
+`getLeaderboards` retrieves list of leaderboards used by your application using specified provider (for example, GameCenter).
```
Comment (borisspektor) [inline src/documents/cocos2dx/cpp/profile/Profile_MainClasses.html.md]: @EugeneButusov please change "retrieves list..." to "retrieves a list..." everywhere

**Comment by EugeneButusov** [comment]
@borisspektor, your feedback was processed.

**Pull request merged**